### PR TITLE
(PUP-9747) Relax validation for bolt

### PR DESCRIPTION
**Pull request by DavidS**
This is an initial draft of the solution. Still needs tests and validation.

### CODE REVIEW

File: lib/puppet/resource_api/transport.rb [hunk 1]
```diff
@@ -53,6 +53,31 @@ def self.validate(name, connection_info)
         environment: current_environment,
       }
     end
+
+    # Attributes we expect from bolt, but want to ignore if the transport does not expect them
+    [:uri, :host, :protocol, :user, :port].each do |attribute_name|
```
Comment (adreyer) [inline lib/puppet/resource_api/transport.rb]: I think we need to add password to this list.

File: lib/puppet/resource_api/transport.rb [hunk 2]
```diff
@@ -53,6 +53,31 @@ def self.validate(name, connection_info)
...
+      if connection_info.key?(attribute_name) && !transports[name].attributes.key?(attribute_name)
+        discard_known_good_attribute(transport_name, attribute_name)
+        connection_info.delete(attribute_name)
+      end
+    end
+
+    # Attributes that bolt emits, but we want to ignore if the transport does not expect them
+    ([:name, :path, :query, :"run-on", :"remote-transport", :implementations] + connection_info.keys.grep { |k| k.to_s.starts_with? 'remote-' }).each do |attribute_name|
```
Comment (adreyer) [inline lib/puppet/resource_api/transport.rb]: for future-proofness I think we need to add `remote-*` to this list.

it's unclear to me how `implementations` ends up here but I don't see any harm in including it.

**Comment by DavidS** [comment]
This still needs to be validated with the bolt tests. I'll get on that next.

**Comment by adreyer** [comment]
I have gotten bolt's test passing with this branch.

**Pull request merged**